src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -285,63 +285,117 @@ def update_app_service_plan(instance, sku=None, number_of_workers=None,
-                                                             backup_request)
-    else:
-        return client.sites.update_site_backup_configuration_slot(resource_group_name, name,
+                                   storage_account_url=storage_account_url, databases=db_setting)
+    if slot:
+        return client.sites.update_site_backup_configuration_slot(resource_group_name, webapp_name,
                                                                   backup_request, slot)
+    else:
+        return client.sites.update_site_backup_configuration(resource_group_name, webapp_name,
+                                                             backup_request)
 
-def restore_backup(resource_group_name, name, storage_account_url, storage_blob_name,
-                   database_name=None, database_type=None, database_connection_string=None,
+def restore_backup(resource_group_name, webapp_name, storage_account_url, backup_name,
+                   db_name=None, db_type=None, db_connection_string=None,
                    target_name=None, overwrite=None, ignore_hostname_conflict=None, slot=None):
     client = web_client_factory()
-    location = _get_location_from_webapp(client, resource_group_name, name)
-    db_setting = DatabaseBackupSetting(database_type, database_name,
-                                       connection_string=database_connection_string)
+    storage_blob_name = backup_name
+    if not storage_blob_name.lower().endswith('.zip'):
+        storage_blob_name += '.zip'
+    location = _get_location_from_webapp(client, resource_group_name, webapp_name)
+    db_setting = _create_db_setting(db_name, db_type, db_connection_string)
     restore_request = RestoreRequest(location, storage_account_url=storage_account_url,
                                      blob_name=storage_blob_name, overwrite=overwrite,
-                                     site_name=target_name, databases=[db_setting],
+                                     site_name=target_name, databases=db_setting,
                                      ignore_conflicting_host_names=ignore_hostname_conflict)
-    if slot is None:
-        return client.sites.restore_site(resource_group_name, name, 0, restore_request)
+    if slot:
+        return client.sites.restore_site(resource_group_name, webapp_name, 0, restore_request, slot)
+    else:
+        return client.sites.restore_site(resource_group_name, webapp_name, 0, restore_request)
+
+def _create_db_setting(db_name, db_type, db_connection_string):
+    if db_name and db_type and db_connection_string:
+        return [DatabaseBackupSetting(db_type, db_name, connection_string=db_connection_string)]
+    elif not db_name and not db_type and not db_connection_string:
+        return None
     else:
-        return client.sites.restore_site(resource_group_name, name, 0, restore_request, slot)
+        raise CLIError('db-name, db-type, and db-connection-string must all be specified together')
```

Comment:
Recommend:
`usage error: --db-name NAME --db-type TYPE --db-connection-string STRING`

This is consistent with argparse usage strings and gives clear direction on what exactly to type.